### PR TITLE
Pla 103 fqn validator fix

### PR DESCRIPTION
**Pull request by AnthonyArdronUW**

### CODE REVIEW

File: business/validation/validation.go [hunk 1]
```diff
@@ -2,9 +2,40 @@
...
+	"strings"
 
-// TODO: Fill up this file as we move code from other codebases here.
+	"github.com/utilitywarehouse/data-infra-mono/business/definitions/catalog/v1"
+)
+
+func ValidateFQNName(fqn string, path string) error {
+	removeSuffix := strings.TrimSuffix(path, ".uwdd.yaml")
+	parts := strings.Split(removeSuffix, "/")
+	targetFQN := fmt.Sprintf("%s/%s/%s", parts[len(parts)-3], parts[len(parts)-2], parts[len(parts)-1])
```
Comment (paulosimao-ardanlabs) [inline business/validation/validation.go:16]: Will we have paths with depth longer than 3? If so, using `strings.ReplaceAll (originalString,".","/")` may be a more reliable way to achieve this.

File: business/validation/validation.go [hunk 2]
```diff
@@ -2,9 +2,40 @@
...
+
+	// must contain versioning as the second element
+	version := parts[len(parts)-2]
```
Comment (paulosimao-ardanlabs) [inline business/validation/validation.go:19]: If this is the case, extracting version with regex may be also safer. 

Please note  that the following regex considers version will be the last portion of the path before the suffixes. I dont remember if we have established standards for these names actually... If not, the easiest would be anyfolder\file.version.uwdd.yaml - IMHO.

```go

package main

import (
	"fmt"
	"regexp"
)

func main() {
	rx := `\.v(\d*).uwdd.yaml`
	r := regexp.MustCompile(rx)
	fmt.Printf("%#v\n", r.FindStringSubmatch(`some.sample.name.v123.uwdd.yaml`)[1])
}

```

**Review by paulosimao-ardanlabs — COMMENTED** [review]
Please see my comments inline

File: business/validation/validation.go (hunk 2 above)
Comment (AnthonyArdronUW) [reply business/validation/validation.go:19]: The version number will appear in a slightly different place but you make a good point.

File: business/validation/validation.go (hunk 1 above)
Comment (AnthonyArdronUW) [reply business/validation/validation.go:16]: The issue i see with that solution is if we want version numbers such as v1.5 the replace will affect it

**Review by paulosimao-ardanlabs — APPROVED** [review]
OK

**Pull request merged**